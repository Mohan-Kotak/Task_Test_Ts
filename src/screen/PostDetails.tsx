import {useRoute} from '@react-navigation/native';
import React from 'react';
import {SafeAreaView, StyleSheet, Text, View} from 'react-native';
import {COLORS} from '../asset/colors';
import {CONSTANTS} from '../asset/constants';

export const PostDetails = () => {
  const route = useRoute<any>();

  return (
    <SafeAreaView>
      <View style={style.mainView}>
        <Text style={style.titleTextStyle}>{route?.params?.item?.title}</Text>
        <View>
          <Text
            style={
              style.aboutTextStyle
            }>{`${CONSTANTS.ABOUT} - ${route?.params?.item?.body}`}</Text>
        </View>
      </View>
    </SafeAreaView>
  );
};

const style = StyleSheet.create({
  mainView: {
    borderWidth: 0.5,
    borderColor: COLORS.GREEN,
    marginTop: 10,
    paddingHorizontal: 10,
    paddingVertical: 10,
  },
  titleTextStyle: {
    fontSize: 22,
    fontWeight: '600',
  },
  aboutTextStyle: {
    fontSize: 14,
    marginTop: 5,
  },
});
