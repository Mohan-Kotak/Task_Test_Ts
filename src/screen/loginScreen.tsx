import {useNavigation} from '@react-navigation/native';
import React, {useState} from 'react';
import {
  Dimensions,
  SafeAreaView,
  StyleSheet,
  Text,
  TouchableOpacity,
  View,
} from 'react-native';
import {COLORS} from '../asset/colors';
import {CONSTANTS} from '../asset/constants';
import {Button} from '../atoms/button';
import {InputText} from '../atoms/inputText';
import auth from '@react-native-firebase/auth';

export const LoginScreen = () => {
  const [email, setEmail] = useState<any>('');
  const [password, setPassword] = useState<any>('');
  const navigation = useNavigation<any>();

  const onSubmitData = () => {
    auth()
      .signInWithEmailAndPassword(email, password)
      .then(() => {
        navigation.navigate('Post', {email});
      })
      .catch(() => {
        console.log('That email address is invalid!');
      });
  };
  return (
    <SafeAreaView style={style.mainView}>
      <View style={style.subView}>
        <View style={style.LoginView}>
          <Text style={style.LoginText}>{CONSTANTS.LOGIN}</Text>
        </View>
        <InputText
          placeholderText={CONSTANTS.EMAIL}
          setText={(ele: any) => setEmail(ele)}
        />
        <InputText
          placeholderText={CONSTANTS.PASSWORD}
          setText={(ele: any) => setPassword(ele)}
          secureText={true}
        />
        <Button buttonText={CONSTANTS.LOGIN} onClick={() => onSubmitData()} />
        <TouchableOpacity
          style={style.LoginView}
          onPress={() => {
            navigation.navigate('SignUp');
          }}>
          <Text style={style.dontHave}>{CONSTANTS.Dont_Have_Account}</Text>
        </TouchableOpacity>
      </View>
    </SafeAreaView>
  );
};

const width = Dimensions.get('window').width;
const height = Dimensions.get('window').height;
const style = StyleSheet.create({
  mainView: {
    width: width,
    height: height,
    paddingHorizontal: 10,
    justifyContent: 'center',
  },
  subView: {
    borderWidth: 1,
    borderColor: COLORS.GREEN,
    paddingHorizontal: 10,
    paddingVertical: 50,
  },
  LoginView: {
    marginBottom: 10,
    alignItems: 'center',
  },
  LoginText: {
    fontSize: 30,
    fontWeight: 'bold',
  },
  dontHave: {
    fontSize: 12,
    fontWeight: 'bold',
  },
});
