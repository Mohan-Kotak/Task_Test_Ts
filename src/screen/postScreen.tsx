import React, {useEffect, useState} from 'react';
import {FlatList, SafeAreaView, StyleSheet, Text, View} from 'react-native';
import axios from 'axios';
import {BASE_URL} from '../service/baseurl';
import {END_URL} from '../service/endurl';
import {useNavigation, useRoute} from '@react-navigation/native';
import {COLORS} from '../asset/colors';
import {CONSTANTS} from '../asset/constants';
import {Button} from '../atoms/button';

export const PostScreen = () => {
  const [postData, setPostData] = useState<any>([]);
  const route = useRoute<any>();
  const navigation = useNavigation<any>();
  console.log(route);
  type Post = {
    userId: number;
    id: number;
    title: any;
    body: any;
  };

  type GetPostResponse = {
    data: Post[];
  };

  useEffect(() => {
    getPost();
  }, []);

  const getPost = () => {
    axios
      .get<GetPostResponse>(`${BASE_URL}${END_URL.POST}`)
      .then(res => {
        setPostData(res?.data);
      })
      .catch(e => {
        console.log(e);
      });
  };

  const renderComponent = ({item}: any) => {
    return (
      <View style={style.mainView}>
        <Text style={style.titleTextStyle}>{item?.title}</Text>
        <View>
          <Text
            style={
              style.aboutTextStyle
            }>{`${CONSTANTS.ABOUT} - ${item?.body}`}</Text>
        </View>
        <Button
          buttonType={CONSTANTS.POST}
          buttonText={CONSTANTS.READ_MORE}
          onClick={() => navigation.navigate('PostDetails', {item})}
        />
      </View>
    );
  };
  return (
    <SafeAreaView>
      <View>
        <Text style={style.name}>{`Hello,${route?.params?.email}`}</Text>
      </View>
      <FlatList
        keyExtractor={item => item?.id}
        data={postData}
        renderItem={renderComponent}
      />
    </SafeAreaView>
  );
};

const style = StyleSheet.create({
  name: {
    fontSize: 16,
    color: COLORS.BLACK,
    marginTop: 10,
  },
  mainView: {
    borderWidth: 0.5,
    borderColor: COLORS.GREEN,
    marginTop: 10,
    paddingHorizontal: 10,
    paddingVertical: 10,
  },
  titleTextStyle: {
    fontSize: 22,
    fontWeight: '600',
  },
  aboutTextStyle: {
    fontSize: 14,
    marginTop: 5,
  },
});
