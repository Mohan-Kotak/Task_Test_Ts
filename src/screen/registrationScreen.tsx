import React, {useState} from 'react';
import auth from '@react-native-firebase/auth';
import {
  Alert,
  Dimensions,
  SafeAreaView,
  StyleSheet,
  Text,
  TouchableOpacity,
  View,
} from 'react-native';
import {COLORS} from '../asset/colors';
import {CONSTANTS} from '../asset/constants';
import {Button} from '../atoms/button';
import {InputText} from '../atoms/inputText';
import {useNavigation} from '@react-navigation/native';

export const RegistationScreen = () => {
  const [email, setEmail] = useState<any>('');
  const [password, setPassword] = useState<any>('');
  const navigation = useNavigation();

  const onSubmitData = () => {
    auth()
      .createUserWithEmailAndPassword(email, password)
      .then(() => {
        Alert.alert('User account created & signed in!');
      })
      .catch(error => {
        if (error.code === 'auth/email-already-in-use') {
          Alert.alert('That email address is already in use!');
        }

        if (error.code === 'auth/invalid-email') {
          Alert.alert('That email address is invalid!');
        }

        Alert.alert(error);
      });
  };

  return (
    <SafeAreaView style={style.mainView}>
      <View style={style.subView}>
        <View style={style.LoginView}>
          <Text style={style.LoginText}>{CONSTANTS.REGISTRATION}</Text>
        </View>
        <InputText
          placeholderText={CONSTANTS.EMAIL}
          setText={(ele: any) => setEmail(ele)}
        />
        <InputText
          placeholderText={CONSTANTS.PASSWORD}
          setText={(ele: any) => setPassword(ele)}
          secureText={true}
        />
        <Button
          buttonText={CONSTANTS.REGISTRATION}
          onClick={() => onSubmitData()}
        />
        <TouchableOpacity
          style={style.LoginView}
          onPress={() => navigation.navigate('Login')}>
          <Text style={style.dontHave}>{CONSTANTS.Have_Login}</Text>
        </TouchableOpacity>
      </View>
    </SafeAreaView>
  );
};

const width = Dimensions.get('window').width;
const height = Dimensions.get('window').height;
const style = StyleSheet.create({
  mainView: {
    width: width,
    height: height,
    paddingHorizontal: 10,
    marginTop: 100,
  },
  subView: {
    borderWidth: 1,
    borderColor: COLORS.GREEN,
    paddingHorizontal: 10,
    paddingVertical: 50,
  },
  LoginView: {
    marginBottom: 10,
    alignItems: 'center',
  },
  LoginText: {
    fontSize: 30,
    fontWeight: 'bold',
  },
  dontHave: {
    fontSize: 12,
    fontWeight: 'bold',
  },
});
