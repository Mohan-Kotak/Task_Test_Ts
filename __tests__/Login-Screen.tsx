import 'react-native';
import React from 'react';
import {LoginScreen} from '../src/screen/loginScreen';
import renderer from 'react-test-renderer';

// for rendering of screen
test('renders correctly', () => {
  const tree = renderer.create(<LoginScreen />).toJSON();
  expect(tree).toMatchSnapshot();
});
