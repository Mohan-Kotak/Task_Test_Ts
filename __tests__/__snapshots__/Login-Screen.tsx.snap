// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<RCTSafeAreaView
  style={
    Object {
      "height": 1334,
      "justifyContent": "center",
      "paddingHorizontal": 10,
      "width": 750,
    }
  }
>
  <View
    style={
      Object {
        "borderColor": "#359934",
        "borderWidth": 1,
        "paddingHorizontal": 10,
        "paddingVertical": 50,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "marginBottom": 10,
        }
      }
    >
      <Text
        style={
          Object {
            "fontSize": 30,
            "fontWeight": "bold",
          }
        }
      >
        Login
      </Text>
    </View>
    <TextInput
      keyboardType="default"
      onChangeText={[Function]}
      placeholder="Email"
      placeholderTextColor="#000000"
      secureTextEntry={false}
      style={
        Object {
          "borderColor": "#359934",
          "borderRadius": 10,
          "borderWidth": 1,
          "height": 50,
          "marginTop": 10,
          "opacity": 0.6,
          "paddingLeft": 20,
        }
      }
    />
    <TextInput
      keyboardType="default"
      onChangeText={[Function]}
      placeholder="Password"
      placeholderTextColor="#000000"
      secureTextEntry={true}
      style={
        Object {
          "borderColor": "#359934",
          "borderRadius": 10,
          "borderWidth": 1,
          "height": 50,
          "marginTop": 10,
          "opacity": 0.6,
          "paddingLeft": 20,
        }
      }
    />
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignSelf": "center",
          "backgroundColor": "#359934",
          "borderRadius": 20,
          "height": 70,
          "justifyContent": "center",
          "marginTop": 10,
          "opacity": 1,
          "width": 200,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#ffffff",
            "fontSize": 22,
            "textAlign": "center",
          }
        }
      >
        Login
      </Text>
    </View>
  </View>
</RCTSafeAreaView>
`;
